server/src/main/java/com/linecorp/centraldogma/server/command/CommandExecutorStatusManager.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.server.command;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.centraldogma.server.management.ServerStatus;
+import com.linecorp.centraldogma.server.management.ServerStatusManager;
+
+/**
+ * Manages the status of a {@link CommandExecutor}.
+ */
+public final class CommandExecutorStatusManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(ServerStatusManager.class);
+
+    private final CommandExecutor executor;
+
+    /**
+     * Creates a new instance.
+     */
+    public CommandExecutorStatusManager(CommandExecutor executor) {
+        this.executor = executor;
+    }
+
+    /**
+     * Returns the executor that this {@link CommandExecutorStatusManager} manages.
+     */
+    public CommandExecutor executor() {
+        return executor;
+    }
+
+    /**
+     * Returns whether the {@link #executor()} is writable.
+     */
+    public boolean writable() {
+        return executor.isWritable();
+    }
+
+    /**
+     * Returns whether the {@link #executor()} is replicating.
+     */
+    public boolean replicating() {
+        return executor.isStarted();
+    }
+
+    /**
+     * Updates the status of the executor with the specified {@link UpdateServerStatusCommand}.
+     *
+     * <p>This method could take a long time if the executor is not in the desired state yet.
+     * So it should be not called from an event loop thread.
+     */
+    public synchronized void updateStatus(UpdateServerStatusCommand command) {
+        final ServerStatus serverStatus = command.serverStatus();
+        setWritable(serverStatus.writable());
+        setReplicating(serverStatus.replicating());
```

Comment:
How about setting replicating first if it changes from false to true?
```suggestion
        final ServerStatus newserverStatus = command.serverStatus();
        if (!replicating() && newserverStatus.replicating()) {
            setReplicating(newserverStatus.replicating());
            setWritable(newserverStatus.writable());
        } else {
            setWritable(newserverStatus.writable());
            setReplicating(newserverStatus.replicating());
        }
```

Reply:
It makes sense. Omitting `!replicating()` condition seems to have the same effect though.